File: src/main/java/com/ptah/common/util/NumberUtils.java
package com.ptah.common.util;

import java.math.BigDecimal;
import java.util.Objects;

public final class NumberUtils {

    public static boolean isGreaterThan(BigDecimal numberA, BigDecimal numberB){
        Objects.requireNonNull(numberA);
        Objects.requireNonNull(numberB);
        return numberA.compareTo(numberB) ==1;
    }

    public static boolean isLessThan(BigDecimal numberA, BigDecimal numberB){
        Objects.requireNonNull(numberA);
        Objects.requireNonNull(numberB);
        return numberA.compareTo(numberB) ==-1;
    }

    public static boolean isGreaterThanOrEqualsTo(BigDecimal numberA, BigDecimal numberB){
        return isGreaterThan(numberA,numberB) || numberA.equals(numberB);
    }
}

File: src/main/kotlin/com/ptah/common/util/NumberUtils.kt
package com.ptah.common.util

import java.math.BigDecimal
import java.util.*

object NumberUtils {
    fun isGreaterThan(numberA: BigDecimal, numberB: BigDecimal): Boolean {
        Objects.requireNonNull(numberA)
        Objects.requireNonNull(numberB)
        return numberA.compareTo(numberB) == 1
    }

    fun isLessThan(numberA: BigDecimal, numberB: BigDecimal): Boolean {
        Objects.requireNonNull(numberA)
        Objects.requireNonNull(numberB)
        return numberA.compareTo(numberB) == -1
    }

    fun isGreaterThanOrEqualsTo(numberA: BigDecimal, numberB: BigDecimal): Boolean {
        return isGreaterThan(numberA, numberB) || numberA == numberB
    }
}
